pxtblocks/blocklycompiler.ts

```diff
@@ -253,6 +257,49 @@ namespace pxt.blocks {
         return ground(check);
     }
 
+    function getReturnTypeOfFunction(e: Environment, name: string) {
+        if (!e.userFunctionReturnValues[name]) {
+            const definition = Blockly.Functions.getDefinition(name, e.workspace);
+
+            const returnTypes: Point[] = [];
+            for (const child of definition.getDescendants(false)) {
+                if (child.type === "function_return") {
+                    attachPlaceholderIf(e, child, "RETURN_VALUE");
+                    returnTypes.push(returnType(e, getInputTargetBlock(child, "RETURN_VALUE")));
+                }
+            }
+
+            let res = mkPoint("void");
+
+            if (returnTypes.length) {
+                try {
+                    const unified = mkPoint(null);
+                    for (const point of returnTypes) {
+                        union(unified, point);
+                    }
+                    res = unified
+                }
+                catch (err) {
+                    e.diagnostics.push({
+                        blockId: definition.id,
+                        message: Util.lf("Function '{0}' has an invalid return type", name)
```

Comment:
I suppose we don't necessarily have enough info here to give a full description of error ("Function can't return both a string and a number" sort of thing), but maybe something along the lines of `"Function '{0}' has an invalid return type; returning different types of values"` -- something to give a description of why it's invalid, in case it's not obvious to the user that `string|number` is making it invalid